images/router/haproxy/reload-haproxy

```diff
@@ -105,7 +105,7 @@ if [ -n "$old_pids" ]; then
     fi
   fi
 
-  /usr/sbin/haproxy -f $config_file -p $pid_file -sf $old_pids
+  /usr/sbin/haproxy -f $config_file -p $pid_file -x /var/lib/haproxy/run/haproxy.sock -sf $old_pids
```

Comment:
Don't we have to start the first process in a different manner also? Is it '-x' then also it is it '-t' for first time start? Need to double check.

Reply:
https://cbonte.github.io/haproxy-dconv/1.8/management.html#3 says:

```
  -x <unix_socket> : connect to the specified socket and try to retrieve any
    listening sockets from the old process, and use them instead of trying to
    bind new ones. This is useful to avoid missing any new connection when
    reloading the configuration on Linux. The capability must be enable on the
    stats socket using "expose-fd listeners" in your configuration.
```

And the config file tells the initial process (and subsequent) where to write the sockets.  So, I think only the reloads need -x.  And the initial doesn't need anything else.

Reply:
BTW I think the right answer is to use the master mode.  But I want to do that when I strip the SYN eater change.  And that doesn't feel like it should be done as a bug fix.